src/DOM/File/Blob.purs

```diff
@@ -23,3 +34,40 @@ type_ blob =
 
 -- | The size (in bytes) of the data contained in the `Blob`.
 foreign import size :: Blob -> Number
+
+-- | An index into the Blob indicating the first byte to include in the new Blob.
+-- | If you specify a negative value, it's treated as an offset from the end of the
+-- | string toward the beginning. For example, -10 would be the 10th from last byte
+-- | in the Blob. If you specify a value for start that is larger than the size
+-- | of the source Blob, the returned Blob has size 0 and contains no data.
+newtype StartByte = StartByte ByteIdx
+
+-- | An index into the Blob indicating the first byte that will *not* be included
+-- | in the new Blob (i.e. the byte exactly at this index is not included).
+-- | If you specify a negative value, it's treated as an offset from the end of
+-- | the string toward the beginning. For example, -10 would be the 10th from
+-- | last byte in the Blob. The default value is size.
+newtype EndByte = EndByte ByteIdx
+
+foreign import data ByteIdx :: Type
+
+-- | Creates `ByteIdx` from `Int` value
+idxFromInt :: Int -> ByteIdx
+idxFromInt = toNumber >>> unsafeCoerce
+
+-- | Creates `ByteIdx` from `Number` value using `Math.round`.
+idxFromNumber :: Number -> ByteIdx
+idxFromNumber = round >>> unsafeCoerce
+
+-- | Creates a new `Blob` object (with specified `MediaType`), containing the
+-- | data in the specified range of bytes of the source Blob, by setting .
+slice ∷ MediaType -> StartByte -> EndByte -> Blob -> Blob
+slice mediaType (StartByte start) (EndByte end) blob =
+  _slice blob start end (unwrap mediaType)
```

Comment:
We could just make this the FFI function directly - there's no need to unwrap any of the newtypes since they have no runtime representation.

Would just need to rearrange the FFI function's argument to match the ones we want :)

Reply:
That's awesome!

Reply:
😆